pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	// Now that basic kubelet configuration is complete, configure networking in the node
+	if err := nc.configureNetwork(); err != nil {
+		return errors.Wrap(err, "configuring node network failed")
+	}
+	log.Info("VM has been configured as a worker node", "VM ID", nc.GetCredentials().GetInstanceId())
+	return nil
+}
+
+// configureNetwork configures k8s networking in the node
+func (nc *nodeConfig) configureNetwork() error {
```

Comment:
I see a lot of code duplication between the validations we have here and in this https://github.com/openshift/windows-machine-config-operator/pull/36/files, @PratikMahajan would you be ok to modularize this into a method called `nc.Validate()`, the code that can go into this would be nc.GetCredentials, IPAddress and other information. This can be part of the PR you are currently working on.